### PR TITLE
feat: regex for ipv6

### PR DESCRIPTION
**Pull request by vasco-santos**
This PR adds regex for `ipv6` addresses per #3 . I needed to do a special validation for addresses started by `0` as they were not being considered `ipv4`. Open to improve this solution

This was based on https://en.m.wikipedia.org/wiki/Reserved_IP_addresses#IPv6

cc @sickcodes @frenchbread

### CODE REVIEW

**Comment by sickcodes** [comment]
I'll give it a test soon! Looks good.

**Comment by sickcodes** [comment]
What was wrong with netmask?

**Comment by vasco-santos** [comment]
`netmask` does not support ipv6 addresses

**Review by sickcodes — APPROVED** [review]
Looks good to me

**Comment by sickcodes** [comment]
```bash
npm run test
168 passed
```

Also tested the inverse of the test and 168 failed which is good.

Here's upstream of is-ip anyway: https://github.com/sindresorhus/ip-regex/blob/master/index.js

@frenchbread

**Comment by frenchbread** [comment]
Thanks for the PR! Good work. Works fine for me as well.

Could you please update the package version also, @vasco-santos & I'll merge it.

**Comment by vasco-santos** [comment]
Sure, added a new commit for the release.

**Comment by frenchbread** [comment]
@vasco-santos Cool, thanks!

File: src/index.js [hunk 1]
```diff
@@ -29,9 +31,27 @@ function netmaskCheck (params) {
   for (let r of privateRanges) {
     if (r.contains(params)) return true
   }
-  return false
```
Comment (frenchbread) [inline src/index.js:32]: Btw, @vasco-santos why did you remove this return statement? Now when I test it by passing public IP it returns `undefiled` rather than `true`.

File: src/index.js [hunk 2]
```diff
@@ -29,9 +31,27 @@ function netmaskCheck (params) {
...
+    /^2001:2[0-9a-fA-F]:([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4})$/.test(params) ||
+    /^2001:db8:([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4})$/.test(params) ||
+    /^2002:([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4}):?([0-9a-fA-F]{0,4})$/.test(params) ||
+    /^f[c-d]([0-9a-fA-F]{2,2}):/i.test(params) ||
+    /^fe[8-9a-bA-B][0-9a-fA-F]:/i.test(params) ||
+    /^ff([0-9a-fA-F]{2,2}):/i.test(params)
+}
+
+export default (ip) => {
+  if (isIp.v4(ip) || ip.startsWith('0')) {
```
Comment (frenchbread) [inline src/index.js:53]: Also, pardon me If I'm misunderstanding, but why add `ip.startsWith('0')` if `netmaskCheck()` already handles it? IMHO we could just have:

```js
export default ip => netmaskCheck(ip) || ipv6Check(ip)
``` 

or 

```js
export default ip => {
  if (isIp.v4(ip)) return netmaskCheck(ip)
  else if (ipIp.v6(ip)) return ipv6Check(ip)
  else return false
}
```

**Comment by frenchbread** [comment]
Sorry, @vasco-santos, for rising these points after code has been merged. I am just wondering for future improvements.

File: src/index.js (hunk 2 above)
Comment (vasco-santos) [reply src/index.js:53]: Yeah, the first was my initial approach. But as I mentioned in the PR description the ipv4 with `0.0.0.0.0` is not a valid ipv4 per the `isIp.v4()`, which is correct... So this should not really be handled in `ipv4`.

The second approach suffer from the same issue I believe. `0.0.0.0.0` will not be nor ipv4, nor ipv6 and it will return false, when it should be true. We can figure out a better solution for such addresses

File: src/index.js (hunk 1 above)
Comment (vasco-santos) [reply src/index.js:32]: Oh this is a bug! I was initially doing everything in the same function, as ipv6 being use d as a fallback for the ipv4 validation. But I moved later. I am going to PR this

File: src/index.js (hunk 2 above)
Comment (frenchbread) [reply src/index.js:53]: I've modified second code snippet:

```js
export default ip => {
  if (isIp.v4(ip) || ip.startsWith('0')) return netmaskCheck(ip)
  else if (isIp.v6(ip)) return ipv6Check(ip)
  else return false
}
```

and getting this:

```
t.truthy(isPrivate(ip), true)
           |         |         
           false     "::ffff:0.0.255.255.255"
-----
t.truthy(isPrivate(ip), true)
           |         |         
           false     "::ffff:0.255.255.255.255"
-----
t.truthy(isPrivate(ip), true)
           |         |         
           false     "100::ffff::"
```

seems like `isIp` is not picking up on those...

Comment (vasco-santos) [reply src/index.js:53]: Yeah, perhaps creating an issue to rethink this logic is a better way
